### ISSUE TITLE
Truly streaming Response from handler?

### ISSUE THREAD

**Issue by eriksunsol** [Starcounter/Home#463]
Starcounter version: `2.3.1.8415`.

### Issue type

- [X] Feature request
- [X] Question

### Issue description

Maybe this is possible already, but if so I could use a pointer to such information, or a code snippet showing an example:

A handler which produces a large payload currently seems only to have the option to generate the entire payload and then pass it to a `Response` object which is in turn returned by the handler. This means that the payload need to be stored temporarily, either in RAM or on disk. The response is however sent to the client through a socket which by nature is a stream so it would be nice if it's streaming nature could be utilized to minimize the amount of temporary storage needed. Also, if the payload takes some time to generate, it looks nicer on the client if data can start transmitting immediately.

E.g. something like this (delegate version):

```CS
Handle.GET("foo", () =>
{
  var resp = new StreamingResponse((Stream stream) =>
    {
      // Code here is called after the handler returns. The stream uses a uses a reasonably
      // sized buffer as a queue. Writes to the stream blocks when the buffer is full. Data
      // from the buffer is written (asynchronously) through the socket connection with
      // the client.
    });

  return resp;
});
```

or like this ("pipe" version):

```C#
Handle.GET("foo", () =>
{
  var resp = new Response() 
  {
    // PipeStream is a read-write stream which uses a uses a reasonably sized buffer
    // acting as a queue.
    // Writes to the stream blocks when the buffer is full. Reads from the
    // stream empties the buffer and blocks if the buffer is empty.
    resp.StreamedBody = new PipeStream();
  }

  Scheduler.RunTask(() =>
  {
    // Code here writes to resp.StreamedBody
    // (The Response object must have some means to determine here that the task
    // has finished writing to the stream.)
  });

  return resp;
});
```

The second example would actually compile just fine, but according to the documentation on `Response.StreamdedBody`, the `Length` property of the stream must be known beforehand, however in this case the whole point is that the size of the payload cannot be known until it has been fully transmitted.

I'm thinking `Transfer-Encoding: chunked` header needs to be used for this...

**Comment by miyconst** [comment]
cc @alemoi

**Comment by miyconst** [comment]
Closing as won't fix in Starcounter 2.4 and not applicable in Starcounter 3.0.
